ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/ip_extcommunity_lists.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from .utils import UtilsMixin
+
+
+class IpExtCommunityListsMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def ip_extcommunity_lists(self) -> dict | None:
+        """
+        Return structured config for ip_extcommunity_lists
+        """
+        if self._overlay_routing_protocol != "ibgp":
+            return None
+
+        if self._vtep_ip is not None:
+            return {
+                "ECL-EVPN-SOO": [
+                    {
+                        "type": "permit",
+                        "extcommunities": f"soo {self._vtep_ip}:1",
+                    },
+                ],
+            }
+
+        return None
```

Comment:
not sure why here you would remove the check for ibgp?

Reply:
this is not a good reasoning though as this will be called both by EBGP and IBGP so we need this first check as `vtep_ip` could be set for EBGP

SOO in extcommunities comes from the templates :) => IDK ;)